### PR TITLE
Add API to support credential update (#1871)

### PR DESCRIPTION
**Pull request by talakad**
* Add API to support credential update

Signed-off-by: veda <veda@portworx.com>

* Review comments

Remove teh checks for google credentials, as its not mandatory

Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

Cherry-pick from master

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -298,6 +298,14 @@ type CredCreateResponse struct {
 	UUID string
 }
 
+// CredUpdateRequest is the input for CredsUpdate command
+type CredUpdateRequest struct {
+	// Name or the UUID of the credential being updated
+	Name string
+	// InputParams is map describing cloud provide
```
Comment (ram-infrac) [inline api/api.go:305]: nit: cloud provider

File: api/server/sdk/credentials.go [hunk 2]
```diff
@@ -589,3 +589,186 @@ func validateAndDeleteIfInvalid(ctx context.Context, s *CredentialServer, uuid s
...
+func (s *CredentialServer) awsUpdate(
+	ctx context.Context,
+	req *api.SdkCredentialCreateRequest,
+	aws *api.SdkAwsCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	params := make(map[string]string)
+
+	params[api.OptCredType] = "s3"
```
Comment (ram-infrac) [inline api/server/sdk/credentials.go:623]: lets make provider type constant

**Review by ram-infrac — APPROVED** [review]
lgtm

File: api/server/sdk/credentials.go (hunk 2 above)
Comment (talakad) [reply api/server/sdk/credentials.go:623]: Since this is a cherry-pick, will leave this as is. lets address this in next PR in master.

**Pull request merged**